### DISCUSSION TITLE
GNOME Shell Extension Pre-release

### DISCUSSION THREAD

**Discussion by andyholmes** [andyholmes/valent#228] [Announcements]
After some consideration, I've decided to open the repository for Valent's [companion GNOME Shell extension][gsx] to the public.

Be warned, this extension has different goals than GSConnect and does not try to be a user interface for Valent. It's sole purpose is to integrate better into GNOME Shell, improving support for:

* Clipboard (especially on Wayland)
* Notifications
* Session Manager (e.g. lock/unlock)

As with Valent, the GNOME Shell extension has nightly builds available in Zip format that can be installed with the `gnome-extensions` tool or another method. There is a thread tracking [breaking changes in the pre-release](https://github.com/andyholmes/valent/discussions/337), although updating the extension at the same time as Valent should be enough.

Have fun!

[gsx]: https://github.com/andyholmes/gnome-shell-extension-valent

**Comment by Baraccc** [comment]
Everyone loves screenshots 🫣

**Reply by andyholmes** [reply]
The UI for the extension is pretty minimal :)

![Screenshot from 2022-08-07 10-42-40](https://user-images.githubusercontent.com/1265208/183304073-6c81f3e0-0a62-46a2-9e75-2be5d2870915.png)

**Comment by edgar-vincent** [comment]
Is it preferable to use this on a laptop running Gnome Shell in lieu of gsconnect when using valent on a Pinephone?
Sorry if my question is silly. I use both an Android phone (with KDEConnect) and a Pinephone (with valent) and would like to find the best setup.

**Reply by andyholmes** [reply]
That's okay, it's good question.

Overall Valent is *probably* less complete but has a newer code base (GTK4, libadwaita, etc). It also has a much more complete test suite, so there's more confidence that things won't explode or break between GNOME releases, but GSConnect has years more real-world testing and thousands of users running it.

The GSConnect Shell extension has a full menu of actions and keyboard shortcuts; Valent's will only ever be a little indicator. Otherwise the features should be the same, but again, newer code base in Valent.

In terms of the service, I have more confidence in Valent although some features are definitely not complete. In particular, the SMS/MMS support is very incomplete, but the PinePhone-side support for that is non-existent so maybe that doesn't matter to you. A few things are more complete in Valent, some are internal that you may not notice. Others like switching audio output (from Android app) and the lock/unlock plugin is supported.

In general I guess GSConnect is still probably the safer bet, but I have not plans to return to that project so without contributions from community members...it is what it is.

**Reply by edgar-vincent** [reply]
Thanks a lot for your reply.
I would in fact like to be able to send and receive my Pinephone SMS from my laptop, so I'm glad that you mentioned the fact that SMS support is non-existent; saves me from looking for them :)

**Reply by andyholmes** [reply]
Yes, that may take awhile. I haven't had a good look into what GNOME Calls supports and there's definitely no cross-desktop interface for that yet.

It's not something I'll put off forever, but my focus is mostly on getting libvalent and the plugin API stable, since ABI/API compatibility is a thing for this project (unlike GSConnect).
